### PR TITLE
Add ORBIT_ prefix to logging-related macros

### PR DESCRIPTION
**Pull request by beckerhe**
This applies what has been discussed in http://go/stadia-orbit-macro-names

- Rename LOG macro to ORBIT_LOG
- Rename ERROR macro to ORBIT_ERROR
- Rename LOG_ONCE macro to ORBIT_LOG_ONCE
- Rename ERROR_ONCE macro to ORBIT_ERROR_ONCE
- Rename FATAL macro to ORBIT_FATAL
- Rename UNREACHABLE macro to ORBIT_UNREACHABLE
- Rename LIKELY macro to ORBIT_LIKELY
- Rename UNLIKELY macro to ORBIT_UNLIKELY
- Rename FAIL_IF macro to ORBIT_FAIL_IF
- Rename CHECK macro to ORBIT_CHECK
- Rename DCHECK macro to ORBIT_DCHECK
- Rename SCOPED_TIMED_LOG_* macros to ORBIT_INTERNAL_SCOPED_TIMED_LOG_*
- Rename SCOPED_TIMED_LOG macro to ORBIT_SCOPED_TIMED_LOG
- Rename PLATFORM_LOG macro to ORBIT_INTERNAL_PLATFORM_LOG
- Rename PLATFORM_ABORT macro to ORBIT_INTERNAL_PLATFORM_ABORT

### CODE REVIEW

**Comment by beckerhe** [comment]
I'm not sure if that's the best way to do it. I had all the different in separate commits, but putting them out in separate PRs would take some time. I assume more than a week until all have been reviewed one after the other. In the end it's literally only renaming things. A two step process is also an option but even more work. Not sure if it's worth it.

Let me know if you prefer to review this differently.

**Review by dpallotti — APPROVED** [review]
I looked at `Logging.h` and quickly looked around, I don't think more is of any use.

> I had all the different in separate commits, but putting them out in separate PRs would take some time.

Just out of curiosity, why didn't you keep the commits separate in this PR? That would have allowed to look at changes for individual macros, for example the ones that are not used a lot (not that it's necessary in this case)

**Review by florian-kuebler — APPROVED** [review]
LGTM, thanks

File: src/OrbitBase/GetLastError.cpp [hunk 1]
```diff
@@ -8,6 +8,8 @@
 #include <absl/strings/ascii.h>
 #include <windows.h>
 
+#include "OrbitBase/Logging.h"
```
[the remark this answers is not in the document]
Comment (beckerhe) [reply src/OrbitBase/GetLastError.cpp:11]: Yes, there was a real bug here hiding. Not a big issue, nevertheless good that we could fix it by this change!

File: src/OrbitBase/Logging.cpp [hunk 2]
```diff
@@ -52,7 +52,7 @@ void InitLogFile(const std::filesystem::path& path) {
   // which tries to lock on the same mutex a second time. This will
   // lead to an error since the mutex is not recursive.
   if (log_file.get() != nullptr) {
-    PLATFORM_ABORT();
+    ORBIT_INTERNAL_PLATFORM_ABORT();
```
[the remark this answers is not in the document]
Comment (beckerhe) [reply src/OrbitBase/Logging.cpp:55]: This came up in the doc. Both `PLATFORM_*` macros are not supposed to be used in the general code base and are just building blocks for the other macros. That's why we opted for marking them internal.

**Comment by beckerhe** [comment]
> I looked at `Logging.h` and quickly looked around, I don't think more is of any use.

I agree. `Logging.h` is the most interesting part.

> 
> > I had all the different in separate commits, but putting them out in separate PRs would take some time.
> 
> Just out of curiosity, why didn't you keep the commits separate in this PR? That would have allowed to look at changes for individual macros, for example the ones that are not used a lot (not that it's necessary in this case)

That's my understanding of our one commit policy. If I wanted to have multiple commits I should have split this up into 14 PRs. I considered this, but that would probably take 2 weeks to get everything in.

File: src/OrbitBase/Logging.cpp (hunk 2 above)
[the remark this answers is not in the document]
Comment (florian-kuebler) [reply src/OrbitBase/Logging.cpp:55]: That makes sense. I thought, I've seen a usage in the general code base here. But I can't find it anymore.

**Comment by dpallotti** [comment]
> That's my understanding of our one commit policy. If I wanted to have multiple commits I should have split this up into 14 PRs. I considered this, but that would probably take 2 weeks to get everything in.

I meant just for the review, and GitHub would have squashed as per "the policy".

**Comment by beckerhe** [comment]
> > That's my understanding of our one commit policy. If I wanted to have multiple commits I should have split this up into 14 PRs. I considered this, but that would probably take 2 weeks to get everything in.
> 
> I meant just for the review, and GitHub would have squashed as per "the policy".

I thought the policy is not only about merging as one commmit, but also about having a single commit for review. There were concerns that having mutliple commits makes the review harder.

**Pull request merged**